crates/node/src/shell/finalize_block.rs

```diff
@@ -470,6 +476,41 @@ where
         None
     }
 
+    // a special event is emitted for each inner tx giving the human readable
+    // name of its wasm code.
+    fn emit_wasm_name_event(
+        &self,
+        tx_data: &TxData<'_>,
+        cmt: &TxCommitments,
+        inner_tx_hash: Hash,
+        response: &mut shim::response::FinalizeBlock,
+    ) {
+        // emit an event that gives the human readable name
+        // of the wasm payload.
+        let code_name = match tx_data
+            .tx
+            .get_section(&cmt.code_hash)
+            .unwrap()
+            .code_sec()
+            .map(|c| c.code.hash())
+        {
+            None => "none".to_string(),
+            Some(hash) => self
+                .state
+                .read(&Key::wasm_code_name(&hash))
+                .ok()
+                .flatten()
+                .unwrap_or("unknown".to_string()),
+        };
```

Comment:
is it better to give up emitting an event in case of `none` or `unknown`? essentially, early return from this fn instead

Reply:
I'm not sure, is it better if the indexer can't find an event for a tx or get one that says it's not clear how to deserialize it?

Reply:
hmm, either way it should be easy to handle in the indexer codebase. I have a slight preference either for not emitting an event at all, or replacing the string with the empty string, to signal absence